### PR TITLE
(readme): make basic example more informative

### LINKED ISSUE #115: onClose not specified in 'Lightbox'
Came across your component today, looks really great! good job!
I tried using it based on your docs and ran into an issue, I get a failed proptypes issue:

> Warning: Failed propType: Required prop `onClose` was not specified in `Lightbox`.

**Steps to reproduce the behavior:**
All i did was follow your instructions given in the DOCS. 

screen shot 2016-12-23 at 14 15 35

### PR DESCRIPTION
**Pull request by goldylucks**
### Motivation
The previous readme is not exhaustive enough and leads to confusion by users, see #115 for example

### Changes
add to quick start example examples of the following:
- onOpen
- onClose
- onClickPrev
- onClickNext

should resolve #115

**Description of changes:**

**Related issues (if any):**

**Checks:**

- [ ] Please confirm `npm run lint` ran successfully
- [ ] Please confirm that only `/src` and `/examples/src` are committed
- [ ] [if new feature] Please confirm that documentation was added to the README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,17 +18,48 @@ import Lightbox from 'react-images';
...
+        />
+      </div>
     );
   }
+
+  onOpen = evt => {
+    this.setState({ lightboxIsOpen: true })
+  }
+
+  onClose = evt => {
```
Comment (tobyweston) [inline README.md]: 😕 should this be `closeLightbox` ❓

**Comment by tobyweston** [comment]
👍

File: README.md [hunk 2]
```diff
@@ -18,17 +18,48 @@ import Lightbox from 'react-images';
...
-        images={[{ src: 'http://example.com/img1.jpg' }, { src: 'http://example.com/img2.jpg' }]}
-        isOpen={this.state.lightboxIsOpen}
-        onClickPrev={this.gotoPrevious}
-        onClickNext={this.gotoNext}
-        onClose={this.closeLightbox}
-      />
+      <div>
+        <button onClick={this.onOpen}>Open</button>
+        <Lightbox
+          currentImage: currentImage
```
Comment (jossmac) [inline README.md]: @goldylucks should this be `currentImage={currentImage}`?

File: README.md [hunk 3]
```diff
@@ -18,17 +18,48 @@ import Lightbox from 'react-images';
...
+          isOpen={lightboxIsOpen}
+          onClickPrev={this.gotoPrevious}
+          onClickNext={this.gotoNext}
+          onClose={this.onClose}
+        />
+      </div>
     );
   }
+
+  onOpen = evt => {
```
Comment (jossmac) [inline README.md]: Please use `myMethod () {}` syntax for:
- onOpen
- onClose
- onClickPrev
- onClickNext

File: README.md [hunk 4]
```diff
@@ -18,17 +18,48 @@ import Lightbox from 'react-images';
...
+    this.setState({ lightboxIsOpen: true })
+  }
+
+  onClose = evt => {
+    this.setState({ lightboxIsOpen: false })
+  }
+
+  onClickPrev = evt => {
+    this.setState({ currentImage: Math.max(0, this.state.currentImage - 1) })
```
Comment (jossmac) [inline README.md]: Please simplify to `this.setState({ currentImage: this.state.currentImage - 1 })`.

File: README.md [hunk 5]
```diff
@@ -18,17 +18,48 @@ import Lightbox from 'react-images';
...
+  }
+
+  onClickNext = evt => {
+    this.setState({
+      currentImage: Math.min(this.state.images.length - 1, this.state.currentImage + 1)
```
Comment (jossmac) [inline README.md]: Please simplify to `this.setState({ currentImage: this.state.currentImage + 1 })`.

**Comment by goldylucks** [comment]
>should this be currentImage={currentImage}?  
yes, fixed

>Please use myMethod () {} syntax  
I'm using fat arrow to bind the context

>Please simplify to ...  
The Min/Max is needed to make sure you don't go to negative index or to index longer than length

**Comment by jossmac** [comment]
@goldylucks cheers mate.

- The min/max is handled by the component internally, safe to remove
- The fat arrow syntax leaves an unused var `evt`, which doesn't help promote clarity. I think it would be more obvious for people reading this to see the `bind` either in the constructor or against the component props.

**Comment by goldylucks** [comment]
1. min / max - gotcha. now that I think about it the entire handling of the current image could be done internally (I can send a PR for that as well)

2. >I think it would be more obvious for people reading the docs to see the bind either in the constructor or against the component props

dunno about that, and it's also best practice to use the fat arrow syntax, so even if that is true it's not a good idea to promote non-best practices. There's no need to call the constructor and add additional boilerplate for bindings.

Thoughts?

**Comment by slim-hmidi** [comment]
@jossmac I tried to make a PR to update the read me after I struggled many problems to function the component `LightBox`, however I discover this opened PR. Does this PR will be added to master or there are something to be fixed. Can I make a PR to update the Read me?

**Comment by jossmac** [comment]
@slim-hmidi I've made some [change-request comments](https://github.com/jossmac/react-images/pull/122/files) that need to be cleaned up before I can merge

**Comment by jossmac** [comment]
@goldylucks you've commented that they're "fixed", did you forget to push those changes?

**Comment by goldylucks** [comment]
@jossmac not sure, it's been a while since then, I don't have the code anymore :)

**Comment by slim-hmidi** [comment]
@jossmac I saw your comments and I reorganize the code and I tested it locally. I could make a pull request to verify it if @jossmac  wants. I will make PR if you want to verify it.

**Comment by davwheat** [comment]
This is a really old PR so I'm going to close it.

If it still applies, please open a new PR with rebased changes.

**Pull request closed without merging**